File: python/isIsomorphic.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        hashMapA = {}
        hashMapB = {}
        for idx in range(len(s)):
            if s[idx] not in hashMapA and t[idx] not in hashMapB:
                hashMapA[s[idx]] = t[idx]
                hashMapB[t[idx]] = s[idx]
            else:
                if (s[idx] in hashMapA and hashMapA[s[idx]] and hashMapA[s[idx]] != t[idx]) or (t[idx] in hashMapB and hashMapB[t[idx]] and hashMapB[t[idx]] != s[idx]):
                    return False
        return True

File: javascript/isIsomorphic.js
function arrayEquals(a, b) {
    return Array.isArray(a) &&
        Array.isArray(b) &&
        a.length === b.length &&
        a.every((val, index) => val === b[index]);
}

var isIsomorphic = function(s, t) {
    let sh = {} ,th = {};
    if (s.length!=t.length)
        return false;
    for (let idx = 0; idx < s.length; idx++) {
        const skey = s[idx];
        const tkey = t[idx];
        if (!(skey in sh )||!(tkey in th)){
            sh[skey] = 1;
            th[tkey] = 1;
            continue;
        }
        sh[skey] ++;
        th[tkey] ++;
    }
    return arrayEquals(Object.values(sh),Object.values(th));
};

console.log(isIsomorphic("egg","add"));